src/visual.ts

```diff
@@ -329,40 +328,59 @@ export class DotPlot implements IVisual {
                     labelFontSize))
             * DotPlot.LabelWidthFactor);
 
-        const maxLabelHeight: number = settings.labels.orientation === DotPlotLabelsOrientation.Vertical
+        const maxLabelHeight: number = settings.labels.orientation.value.value === DotPlotLabelsOrientation.Vertical
             ? maxLabelWidth
             : 0;
 
-        const radius: number = settings.dataPoint.radius;
+        const radius: number = settings.dataPoint.radius.value;
 
         const diameter: number = DotPlot.RadiusFactor * radius + DotPlot.ExtraDiameter,
             dotsTotalHeight: number = height - maxXAxisHeight
                 - radius * DotPlot.RadiusFactor - labelFontSize - layout.margin.top - maxLabelHeight,
             maxDots: number = Math.floor(dotsTotalHeight / diameter);
 
-        const yScale: LinearScale<number, number> = d3.scaleLinear()
+        const yScale: d3LinearScale<number, number> = d3ScaleLinear()
             .domain([DotPlot.MinAmountOfDots, maxDots])
             .range([dotsTotalHeight, DotPlot.MinAmountOfDots]);
 
-        const dataPointsGroup: DotPlotDataGroup[] = [],
-            minDots = minValue / (maxValue / maxDots),
-            additionalValue = minValue <= 1 ? -minValue + 1 : 0; // negative values scales incorrect
+        const minDots = minValue / (maxValue / maxDots);
+        const additionalValue = minValue <= 1 ? -minValue + 1 : 0; // negative values scales incorrect
 
-        const dotScale: LogScale<number, number> = d3.scaleLog()
+        const dotScale: d3LogScale<number, number> = d3ScaleLog()
             .domain(DotPlot.getDomain(minValue + additionalValue, maxValue + additionalValue))
             .range(DotPlot.getDomain(minDots > maxDots ? 1 : minDots, maxDots))
             .clamp(true);
 
+        const dataPointsGroup: DotPlotDataGroup[] = DotPlot.generateDotPlotDataGroup(valueValues, dotScale, additionalValue, valueColumn, maxDots, yScale, settings, localizationManager, visualHost, categoryColumn, categories, formattedValues, labelFontSize);
+
+        return {
+            categoryColumn,
+            settings,
+            maxXAxisHeight,
+            labelFontSize,
+            dotsTotalHeight,
+            maxLabelWidth,
+            maxLabelHeight,
+            maxCategoryWidth,
+            dataGroups: dataPointsGroup,
+            categoryAxisName: categoryColumn.source.displayName,
+            categoryLabelHeight: DotPlot.DefaultCategoryLabelHeight
+        };
+    }
+
+    private static generateDotPlotDataGroup(valueValues: number[], dotScale: d3LogScale<number, number, never>, additionalValue: number, valueColumn: powerbi.DataViewValueColumn, maxDots: number, yScale: d3LinearScale<number, number, never>, settings: DotPlotSettingsModel, localizationManager: ILocalizationManager, visualHost: IVisualHost, categoryColumn: powerbi.DataViewCategoryColumn, categories: DotPlotChartCategory[], formattedValues: string[], labelFontSize: number) {
```

Comment:
too much arguments, please use object as argument.

Reply:
Done.

Reply:
Don't see a change here. Have you pushed a commit?